module/VuFind/src/VuFind/Config/Feature/ConfigSettingPropertiesTrait.php

```diff
@@ -0,0 +1,252 @@
+    ): array {
+        return $this->localizableSettings[$contextKey];
+    }
+
+    /**
+     * Is the setting localizable?
+     *
+     * @param string $setting    Setting
+     * @param string $contextKey Key identifying the context (optional)
+     *
+     * @return bool
+     */
+    public function isLocalizableSetting(
+        string $setting,
+        string $contextKey = self::DEFAULT_CONTEXT
+    ): bool {
+        return in_array($setting, $this->getLocalizableSettings($contextKey));
+    }
+
+    /**
+     * Validate settings.
+     *
+     * @param array  $settings   Settings
+     * @param string $contextKey Key identifying the context (optional)
+     *
+     * @return array
+     * @throws BadConfig
+     */
+    public function validateSettings(
+        array $settings,
+        string $contextKey = self::DEFAULT_CONTEXT
+    ): array {
+        foreach ($this->getRequiredSettings($contextKey) as $requiredSetting) {
+            if (
+                !isset($settings[$requiredSetting])
+                && $this->isRequiredSetting($requiredSetting, $settings, $contextKey)
+            ) {
+                throw new BadConfig(
+                    'Missing required setting: ' . $requiredSetting
+                );
+            }
+        }
+        return $settings;
+    }
+
+    /**
+     * Localize the settings if possible.
+     *
+     * @param array  $settings        Settings to localize
+     * @param string $userLocale      User locale
+     * @param array  $fallbackLocales Fallback locale(s) (optional)
+     * @param string $contextKey      Key identifying the context (optional)
+     * @param bool   $useFirst        Use first array item if item matching
+     *                                locale(s) was not found (optional)
+     *
+     * @return array
+     */
+    public function localizeSettings(
+        array $settings,
+        string $userLocale,
+        array $fallbackLocales = [],
+        string $contextKey = self::DEFAULT_CONTEXT,
+        bool $useFirst = true
+    ): array {
+        foreach ($settings as $key => $value) {
+            $settings[$key]
+                = $this->localizeSetting(
+                    $key,
+                    $value,
+                    $userLocale,
+                    $fallbackLocales,
+                    $contextKey,
+                    $useFirst
+                );
+        }
+        return $settings;
+    }
+
+    /**
+     * Localize the setting if possible.
+     *
+     * @param string       $key             Key
+     * @param string|array $value           Value
+     * @param string       $userLocale      User locale
+     * @param array        $fallbackLocales Fallback locale(s) (optional)
+     * @param string       $contextKey      Key identifying the context (optional)
+     * @param bool         $useFirst        Use first array item if item matching
+     *                                      locale(s) was not found (optional)
```

Comment:
Can unwrap here too:

```suggestion
     * @param bool         $useFirst        Use first array item if item matching locale(s) was not found (optional)
```